src/components/ProcessImage.js

```diff
@@ -114,26 +119,45 @@ class ProcessImage extends Component {
   myStorage = null;
 
   processInMainThread = props => {
-    ROOT.read(props.image).then(image => {
-      processImage(image, props, ROOT).getBase64(ROOT.AUTO, (err, src) => {
-        if (this.state.src !== src || this.state.err !== err) {
-          this.setState({ src, err });
-          this.passPropsToParent(props, src, err);
-          if (typeof props.onProcessFinish === 'function') {
-            props.onProcessFinish();
+    ROOT.read(props.image)
+      .then(image => {
+        processImage(image, props, ROOT).getBase64(ROOT.AUTO, (err, src) => {
+          if (this.state.src !== src || this.state.err !== err) {
+            this.setState({ src, err });
+            this.passPropsToParent(props, src, err);
+            if (typeof props.onProcessFinish === 'function') {
+              props.onProcessFinish();
+            }
           }
+        });
+      })
+      .catch(err => {
+        this.passPropsToParent(props, '', result.err);
+        if (typeof props.onProcessFinish === 'function') {
+          props.onProcessFinish();
         }
+        return;
       });
-    });
   };
 
   processInWebWorker = async (worker, props, storageReference) => {
     if (worker !== null) {
-      const result = await worker.process({
-        props: filterPropsToListen(props),
-        image: props.image,
-        customCdn: props.customCdn
-      });
+      const result = await worker
+        .process({
+          props: filterPropsToListen(props),
+          image: props.image,
+          customCdn: props.customCdn
+        })
+        .catch(err => ({ err }));
+
+      if (typeof result.src === 'undefined' && result.err) {
```

Comment:
I created this if statement to break out of an infinite loop. componentDidUpdate calls this method and the image would try to be loaded again causing another error.